lookup_generation/nwis_lookups.py

```diff
@@ -0,0 +1,64 @@
+"""
+Utilities to retrieve NWIS RDB codes
+"""
+
+from itertools import groupby
+
+
+def _get_lookup_value(code_lookup, name_key, desc_key):
+    """
+    Returns dict with 'name' and optional 'desc' properties derived from code_lookup
+    :param code_lookup:
+    :param name_key:
+    :param desc_key: Can be the null string
+    :rtype: dict
+    """
+    lookup_value = {'name': code_lookup.get(name_key)}
+    if desc_key:
+        lookup_value['desc'] = code_lookup.get(desc_key)
+    return lookup_value
+
+
+def translate_to_lookup(dict_iter, code_key, name_key, desc_key):
+    """
+    Translate dict_iter to a single dictionary where the code_key values will be be the keys with
+    the value a dictionary with 'name' and 'desc' properties.
+    :param iter dict_iter:
+    :param str code_key: Should not be null
+    :param str name_key: Should not be null
+    :param str desc_key: Can be the null string
+    :rtype: dict
+    """
+
+    def has_code_key(lookup):
+        return code_key in lookup
+
+    filtered_dict = filter(has_code_key, dict_iter)
```

Comment:
I'm wondering if the code would be more concise if `has_code_key` is rewritten as a lambda function... :thinking: